internal/validator/default/default_validator.go

```diff
@@ -7,25 +7,29 @@ import (
 	"github.com/bitcoin-sv/arc/internal/validator"
 	"github.com/bitcoin-sv/arc/pkg/api"
 	"github.com/libsv/go-bt/v2"
-	"github.com/libsv/go-bt/v2/bscript/interpreter"
 	"github.com/ordishs/go-bitcoin"
 )
 
 type DefaultValidator struct {
-	policy *bitcoin.Settings
+	policy   *bitcoin.Settings
+	txFinder validator.TxFinderI
 }
 
-func New(policy *bitcoin.Settings) *DefaultValidator {
+func New(policy *bitcoin.Settings, finder validator.TxFinderI) *DefaultValidator {
 	return &DefaultValidator{
-		policy: policy,
+		policy:   policy,
+		txFinder: finder,
 	}
 }
 
 func (v *DefaultValidator) ValidateTransaction(ctx context.Context, tx *bt.Tx, feeValidation validator.FeeValidation, scriptValidation validator.ScriptValidation) error { //nolint:funlen - mostly comments
 	// 0) Check whether we have a complete transaction in extended format, with all input information
 	//    we cannot check the satoshi input, OP_RETURN is allowed 0 satoshis
-	if !v.IsExtended(tx) {
-		return validator.NewError(fmt.Errorf("transaction is not in extended format"), api.ErrStatusTxFormat)
+	if needsExtention(tx, feeValidation, scriptValidation) {
```

Comment:
If a tx will only be extended if either a fee or script validation happens it may be good to have a small e2e test which ensures that sending a **standard** format tx while **skipping script and fee validation** still works.

A template for that can be found here: https://github.com/bitcoin-sv/arc/blob/2341e4ae688b05e32f1d05d16aeae2e4f712b707/test/endpoint_test.go#L317:L376

Reply:
New two tests have been added.